### DISCUSSION TITLE
Setting Up Debugging of PAI (Issues with MG5050+)

### DISCUSSION THREAD

**Discussion by afblom** [ParadoxAlarmInterface/pai#407] [General]
Hey all,
I am getting connection issues with the MG5050+ panel when connecting to PAI via serial. The symptoms are a bit strange but viewing the most details logs, I can see that it will make an innitial connection and get the Device data, followed by sometimes getting at least the initial setup of my home. But it then looses comms and is never able to recover. 

I am an experience C programmer but python is something I have dabbled with for scripts only. Is there any advice for setting up this python project (never worked on such a big python prj) for debugging to try and spot what would be causing the aforementioned issue? I was also wondering if anyone has access to the serial protocol documents, this could also be something I could use to validate the communucation with the Panel (but for the life of me cannot find it online)

Or of course! If anyone has any sugguestions on what may  be causing the issue I mentioned, I would be super stoked to hear! :)

**Comment by yozik04** [comment]
Hey @afblom. So far how much I tried I have failed to figure out why it disconnects. Nothing criminal on our side. Panel just stops responding after some time. PAI detects that and reconnects. In "normal" situations this happens infrequently.
I see in your case something is off.
There is no serial protocol documentation available. All is reverse engineered.

To start with, you can set
```
LOGGING_DUMP_PACKETS = True
LOGGING_LEVEL_CONSOLE = logging.DEBUG
```
